### PR TITLE
Add json config schema

### PR DESCRIPTION
**Pull request by hreinecke**
Add a schema file to describe the JSON configuration file format.

Signed-off-by: Hannes Reinecke <hare@suse.de>

### CODE REVIEW

**Comment by keithbusch** [comment]
Looks good to me

Reviewed-by: Keith Busch <kbusch@kernel.org>

File: doc/config-schema.json [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+		    "description": "Enable data digest",
+		    "type": "boolean",
+		    "default": false
+		},
+		"persistent": {
+		    "description": "Create persistent discovery connection",
+		    "type": "boolean"
+		}
+	    },
+	    "required": [ "transport" ]
```
Comment (sagi-lb) [inline doc/config-schema.json:140]: traddr is also mandatory no?

File: doc/config-schema.json [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		    "type": "integer"
+		},
+		"ctrl_loss_tmo": {
+		    "description": "Controller loss timeout (in seconds)",
+		    "type": "integer"
+		},
+		"fast_io_fail_tmo": {
+		    "description": "Fast I/O Fail timeout (in seconds)",
+		    "type": "integer",
+		    "default": 600
```
Comment (sagi-lb) [inline doc/config-schema.json:108]: why?

File: doc/config-schema.json (hunk 1 above)
Comment (hreinecke) [reply doc/config-schema.json:140]: Not for pcie.

File: doc/config-schema.json (hunk 2 above)
Comment (hreinecke) [reply doc/config-schema.json:108]: Because that's what the code does. No specific reason otherwise.

**Pull request merged**